### PR TITLE
internal/charmstore: fix search

### PR DESCRIPTION
**Pull request by mhilton**
Update the search queries and indexes to reduce the number of irrelevant results returned in searches.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1769/
Test PASSed.

**Comment by rogpeppe** [comment]
LGTM but I'd like to see this end up as two commits - one that just adds the test cases (and some ExpectFailures) and another that applies the actual fix. That way we can go back to the "fix" commit some time later and easily see what how fix fixed the issue.

File: internal/charmstore/elasticsearch.go [hunk 1]
```diff
@@ -49,313 +49,285 @@ const esIndexJSON = `
...
+            "omit_norms": true,
+            "index_options": "docs"
           },
-          "ngrams" : {
-            "type" : "string",
-            "analyzer" : "n3_20grams",
-            "include_in_all" : false
+          "ngrams": {
+            "type": "string",
+            "index_analyzer": "n3_20grams",
```
Comment (jrwren) [inline internal/charmstore/elasticsearch.go:88]: This change from analyzer to index_analyzer, does it mean it was defaulting to some other analyzer and we never were using n3_20grams on this field?

**Comment by jrwren** [comment]
👍

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1770/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1771/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1772/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1773/
Test PASSed.

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**